ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTablePostgresql.scala

```diff
@@ -0,0 +1,231 @@
+      insertEvents.foreach(_.execute())
+      updateArchives.foreach(_.execute())
+    }
+  }
+
+  private val updateArchived =
+    """update participant_events set create_consumed_at={consumed_at} where contract_id={contract_id} and create_argument is not null"""
+
+  private def archive(consumedAt: Offset)(contractId: ContractId): Vector[NamedParameter] =
+    Vector[NamedParameter](
+      "consumed_at" -> consumedAt,
+      "contract_id" -> contractId.coid,
+    )
+
+  override def toExecutables(
+      tx: TransactionIndexing.TransactionInfo,
+      info: TransactionIndexing.EventsInfo,
+      serialized: TransactionIndexing.Serialized,
+  ): EventsTable.Batches = {
+    val capacity = info.events.size
+    val eventIds = Array.fill(capacity)(null.asInstanceOf[String])
+    val eventOffsets = Array.fill(capacity)(tx.offset.toByteArray)
+    val contractIds = Array.fill(capacity)(null.asInstanceOf[String])
+    val transactionIds = Array.fill(capacity)(tx.transactionId.asInstanceOf[String])
+    val workflowIds = Array.fill(capacity)(tx.workflowId.map(_.asInstanceOf[String]).orNull)
+    val ledgerEffectiveTimes = Array.fill(capacity)(tx.ledgerEffectiveTime)
+    val templateIds = Array.fill(capacity)(null.asInstanceOf[String])
+    val nodeIndexes = Array.fill(capacity)(null.asInstanceOf[Int])
+    val commandIds =
+      Array.fill(capacity)(tx.submitterInfo.map(_.commandId.asInstanceOf[String]).orNull)
+    val applicationIds =
+      Array.fill(capacity)(tx.submitterInfo.map(_.applicationId.asInstanceOf[String]).orNull)
+    val submitters = Array.fill(capacity)(
+      tx.submitterInfo.map(_.singleSubmitterOrThrow().asInstanceOf[String]).orNull)
+    val flatEventWitnesses = Array.fill(capacity)(null.asInstanceOf[String])
+    val treeEventWitnesses = Array.fill(capacity)(null.asInstanceOf[String])
+    val createArguments = Array.fill(capacity)(null.asInstanceOf[Array[Byte]])
+    val createSignatories = Array.fill(capacity)(null.asInstanceOf[String])
+    val createObservers = Array.fill(capacity)(null.asInstanceOf[String])
+    val createAgreementTexts = Array.fill(capacity)(null.asInstanceOf[String])
+    val createConsumedAt = Array.fill(capacity)(null.asInstanceOf[Array[Byte]])
+    val createKeyValues = Array.fill(capacity)(null.asInstanceOf[Array[Byte]])
+    val exerciseConsuming = Array.fill(capacity)(null.asInstanceOf[Boolean])
+    val exerciseChoices = Array.fill(capacity)(null.asInstanceOf[String])
+    val exerciseArguments = Array.fill(capacity)(null.asInstanceOf[Array[Byte]])
+    val exerciseResults = Array.fill(capacity)(null.asInstanceOf[Array[Byte]])
+    val exerciseActors = Array.fill(capacity)(null.asInstanceOf[String])
+    val exerciseChildEventIds = Array.fill(capacity)(null.asInstanceOf[String])
+
+    for (((nodeId, node), i) <- info.events.zipWithIndex) {
+      node match {
+        case create: Create =>
+          val eventId = EventId(tx.transactionId, nodeId)
```

Comment:
Why not inline this?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8042/commits/bb952a2afc862305d03dcc8060566df5114fca2b